scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,435 @@
+package org.oppia.android.scripts.maven
+
+import org.oppia.android.scripts.maven.data.CopyrightLicense
+import org.oppia.android.scripts.maven.data.Dependency
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+import java.io.File
+import java.io.FileInputStream
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.Transformer
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+private const val MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE =
+  "maven_dependencies.textproto is not up-to-date"
+
+/**
+ * Script to extract the licenses for the third-party Maven dependencies (direct and indirect both)
+ * on which Oppia Android depends.
+ *
+ * Usage:
+ *   bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+ *     <path_to_maven_dependenices.pb>
+ *
+ * Arguments:
+ * - path_to_directory_values: directory path to the values folder of the Oppia Android repository.
+ * - path_to_maven_dependenices.pb: relative path to the maven_dependencies.pb
+
+ * Example:
+ *   bazel run //scripts:generate_license_texts -- $(pwd)/app/src/main/res/values
+ *   scripts/assets/maven_dependencies.pb
+ */
+fun main(args: Array<String>) {
+  GenerateLicenseTexts(LicenseFetcherImpl()).main(args)
+}
+
+/**
+ * Wrapper class to pass dependencies to be utilized by the the main method to generate license
+ * texts of the dependencies.
+ */
+class GenerateLicenseTexts(
+  private val licenseFetcher: LicenseFetcher
+) {
+
+  /** Generates a resource xml file that contains license texts of the third-party dependencies. */
+  fun main(args: Array<String>) {
+    if (args.size < 2) {
+      println(
+        """
+        Usage: bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+        <path_to_pb_file>  
+        """.trimIndent()
+      )
+      throw Exception("Too few arguments passed.")
+    }
+    val pathToValuesDirectory = args[0]
+    val pathToThirdPartyDependenciesXml = "$pathToValuesDirectory/third_party_dependencies.xml"
```

Comment:
Suggest using File to compute a more robust path (to avoid depending on the argument to be passed a certain way:

```suggestion
    val valuesDirectory = File(pathToValuesDirectory) // Maybe verify this exists & is a directory with a check?
    val thirdPartyDependenciesXml = File(valuesDirectory, "third_party_dependencies.xml")
```

Ditto for other files (if applicable) to simplify & improve robustness.

Reply:
Done.